index.html

```diff
@@ -1268,6 +1271,17 @@ <h3>Types</h3>
           </thead>
 
           <tbody>
+            <tr>
+              <td>
+<a>verifiable credential</a>&nbsp;object <br />(a subclass of <a href="#credentials">Credential</a>&nbsp;object)
+              </td>
+              <td>
+<code>VerifiableCredential</code> and a more specific
+<a>verifiable credential</a> <a>type</a>. For example,<br>
+<code>"type": ["VerifiableCredential", "UniversityDegreeCredential"]</code>
```

Comment:
We don't have a hard requirement to have more than one type for a credential or a presentation; the only MUST is that there is *at least* `VerifiableCredential` on a VC and *at least* `VerifiablePresentation` on a VP. The above change might make people think you must have two types on each. I've suggested `optionally` in both cases to address this.

Reply:
Well, apparently our spec *does* say this is required at present. I feel like that is a mistake, but one that can be lived with. It likely creates an undue burden on VPs, though on VCs it's sensible. I need to check to see how long that's been in the spec or if we added as an attempt to clarify and made a mistake.

Reply:
Looks like this change brought in that requirement on VPs and encapsulated objects:

https://github.com/w3c/vc-data-model/commit/30f8bfc338ff3c9449e41a78a8c09133dbe79751

But it left open the door to omitting specifying this information with this clause (emphasis mine):

> All credentials, presentations, and encapsulated objects MUST specify, ***or be associated with***, additional more narrow types...

I'm not sure how to interpret "or be associated with", but we should clarify. One interpretation is that VCs and VPs should contain enough sufficiently typed information to allow for processing. I don't think the intent was to ever require narrow types for every VP and encapsulated object and it seems that clause allows for wiggle room there. Prior to this change there was a requirement for a more narrow type for credentials, again, which is sensible, but even that, I don't think, should be strictly necessary.

Reply:
I believe I agree... Merging your tweaks.

Reply:
@dlongley "We don't have a hard requirement to have more than one type for a credential or a presentation"

I believe we do for a verifiable credential at least. The type parameter is there to tell the JSON parser what type of VC this is so that it knows what properties to expect. The credentialSchema property further says which properties are mandatory and which are optional.

The phrase "or be associated with" was there so that encapsulated objects did not have to have their types in the objects themselves, but they could be in the encapsulating object so that the parser could sooner determine if it can process this type of VC or not.

We had several long discussions about whether VCs should be strongly typed or not and if you look through the minutes you will see that strongly typed was agreed.

Reply:
@David-Chadwick, as stated above, I'm less concerned about this for VCs and it's generally good practice to include a more narrow type. This is because there's an expectation that extensions will be created to provide the necessary terms to express whatever is being claimed about credential subject(s).

However, VPs are quite usable without such extensions. I think requiring an extra type for VPs is a problem that would just ultimately result in the creation of an additional, generally useless (semantically irrelevant) type that people tack onto their VPs just to make them conformant.